lib/translator/command_translator.rb

```diff
@@ -93,10 +95,19 @@ def kernel_builder
 
             # --- Actual Visitor parts stars here ---
 
+            def visit_array_command(command)
+                if command.keep && command.gpu_result_pointer.nil?
+                    # Create slot for result pointer on GPU in env
+                    environment_builder.add_previous_result(command.unique_id, 0)
```

Comment:
Create instance method `allocate_previous_pointer` without the second parameter (uses `0`) internally. Maybe a bit cleaner.